### PR TITLE
Fix country specific search

### PR DESCRIPTION
**Pull request by knirpsfass**
In this PR i fixed the country search since the new ui for the playstore introduced a **About the results** button for european countries.

Also i've changed the **more results** logic to no longer use a non localized string and isntead check if the value on a specific path is a string

For further information about the issue check: https://github.com/facundoolano/google-play-scraper/issues/551 and  #563

### CODE REVIEW

**Review by facundoolano — CHANGES_REQUESTED** [review]
Can you add a unit test to cover this case?

File: lib/search.js [hunk 1]
```diff
@@ -118,9 +123,16 @@ async function processFirstPage (html, opts, savedApps, mappings) {
...
+  const sectionTitle = R.path(SECTIONS_MAPPING.title, section);
+  return R.is(String, sectionTitle) && !R.isEmpty(sectionTitle);
+}
+
 const INITIAL_MAPPINGS = {
   app: ['ds:4', 0, 1, 0, 23],
-  sections: ['ds:4', 0, 1]
+  sections: ['ds:4', 0, 1],
+  aboutResultSection: ['ds:4', 0, 1],
+  aboutResultsTitle: ['ds:4', 0, 1, 0, 31, 0]
```
Comment (mmoksh) [inline lib/search.js:135]: It looks like `aboutResultsTitle` has the same mapping as `sections`. Why do you have to duplicate the element?

File: lib/search.js [hunk 2]
```diff
@@ -118,9 +123,16 @@ async function processFirstPage (html, opts, savedApps, mappings) {
   return checkFinished(opts, [...savedApps, ...apps], token);
 }
 
+function isMoreSection (section) {
+  const sectionTitle = R.path(SECTIONS_MAPPING.title, section);
+  return R.is(String, sectionTitle) && !R.isEmpty(sectionTitle);
```
Comment (mmoksh) [inline lib/search.js:128]: The first non-empty title does not guarantee it's the more_results section. Try a random query like `lajsdfkljew`. The first section should have a title: `No results for lajsdfkljew`. I think this case will break the script.

File: lib/search.js (hunk 1 above)
Comment (mmoksh) [reply lib/search.js:135]: Also, the `aboutResultsTitle` should be part of a SECTION mapping, not the initial mappings. It's something specific to one section.

File: lib/search.js (hunk 2 above)
Comment (knirpsfass) [reply lib/search.js:128]: This would return
```
{
    "message": "Keine Ergebnisse für <b>lajsdfkljew</b>"
}
```

File: lib/search.js (hunk 1 above)
Comment (knirpsfass) [reply lib/search.js]: as suggested its now using the section mapping + the result title mapping was moved to section specific mapping

**Comment by knirpsfass** [comment]
> Can you add a unit test to cover this case?

i'll add tests asap

**Comment by mmoksh** [comment]
Would you please add the following tests and make sure the code change doesn't break them?

1. query = "netflix"
=> expect(results[0] = "com.netflix.mediaclient")
=> expect("com.netflix.ninja" in results)
=> expect("?id=com.netflix.NGP.StrangerThings" in results)

2. query = "gmail"
=> expect(results[0] = "com.netflix.mediaclient")
=> expect(X in results) >> basically, make sure that the script reads the "more results" section, not the "similar apps" section

3. query = "netflix", hl=de, gl=de
=> expect(results[0] = "com.netflix.mediaclient")
=> expect("com.netflix.ninja" in results)
=> expect("?id=com.netflix.NGP.StrangerThings" in results)

4. query = "games"
=> expect("com.kiloo.subwaysurf" in results)

5. query = "kfdsjlkdsjifoewdsk"
=> expect empty set (currently it throws an error - we can change this behaviour)

**Comment by knirpsfass** [comment]
@mmoksh am i mistaken or is the empty set not throwing an error on the master?

File: lib/search.js (hunk 2 above)
Comment (knirpsfass) [reply lib/search.js:128]: Okay never mind I'm kinda stupid😅. So yeah it still throws as it did before

**Comment by ChayaGel** [comment]
UnhandledPromiseRejectionWarning: Error: No results for <b>cfgduyhgfuyjbgfuyhfuy</b>

**Comment by knirpsfass** [comment]
@ChayaGel is this really an issue of this version?
With 9.0.0 the api no longer returns an empty array instead it throws - see https://github.com/facundoolano/google-play-scraper/issues/566#issue-1286959224

**Comment by ChayaGel** [comment]
ok sorry

**Pull request merged**